_posts/2021-03-23-connector-build.md

```diff
@@ -0,0 +1,232 @@
+* In case you use Openshift, the image will be built by Openshift Builds
+
+In both cases, Strimzi will spin-up a build pod, which builds the image based on the configuration from custom resource.
+The final image is then pushed into specific registry or image stream, again based on the configuration. 
+The great thing in this feature is, that you can easily reuse the newly built image in another KafkaConnect resource without building it again or manually.
+
+### KafkaConnect configuration
+A new `build` configuration for KafkaConnect resource allow you to configure a list of custom connectors, which will be downloaded and baked into a new KafkaConnect images specified by you.
+
+In the example bellow, you can see the configuration of the build part in KafkaConnect:
+* (1) - build configuration which contains output information and list of plugins.
+* (2) - configuration of registry, where new image will be pushed. 
+* (3) - List of plugins, which will be downloaded and added into your specific connect image.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+  annotations:
+    strimzi.io/use-connector-resources: "true"
+spec:
+  ...
+  build: (1)
+    output: (2)
+      type: docker
+      image: my-registry.io/my-org/my-connect-cluster:latest
+      pushSecret: my-registry-credentials
+    plugins: (3) 
+      - name: debezium-postgres-connector
+        artifacts:
+          - type: tgz
+            url: https://repo1.maven.org/maven2/io/debezium/debezium-connector-postgres/1.4.2.Final/debezium-connector-postgres-1.4.2.Final-plugin.zip
+            sha512sum: ef1620547e6ddf5be010271849b6a87a19f6e6beee93b379c80883815b8f37ec5137095b2c99975d7704cbf957e6a33d76c61109582cad57c7cbbfae43adc86c
+      - name: camel-timer
+        artifacts:
+          - type: tgz
+            url: https://repo.maven.apache.org/maven2/org/apache/camel/kafkaconnector/camel-timer-kafka-connector/0.8.0/camel-timer-kafka-connector-0.8.0-package.tar.gz
+            sha512sum: c0102700ae176b1ef078bdb86d640935a1333865e58b870f6b0ef3d9dad067221122aee6bf52852dad209a7c2051f359e0434fc7fbd783fb50e21707c7577ef9
+      - name: echo-connector
+        artifacts:
+          - type: jar
+            url: https://github.com/scholzj/echo-sink/releases/download/1.1.0/echo-sink-1.1.0.jar
+            sha512sum: b7da48d5ecd1e4199886d169ced1bf702ffbdfd704d69e0da97e78ff63c1bcece2f59c2c6c751f9c20be73472b8cb6a31b6fd4f75558c1cb9d96daa9e9e603d2
+```
+
+There is a several possible options which you can configure for the build.
+In `output` part, you need to specify output type either to `docker` for push into docker registry like quay.io or to `imagestream` to push image into internal Openshift image stream (openshift only).
+Output configuration also require image name and push secret in case the registry are private. 
+The secret has to be deployed in the same namespace as KafkaConnect resource.
```

Comment:
Maybe you can explain or add link to Kubernetes docs to make it clear how the secret should look like?

Reply:
Done